java/runtime/src/main/java/org/ray/runtime/object/RayObjectImpl.java

```diff
@@ -57,9 +54,25 @@ protected void finalize() throws Throwable {
     }
   }
 
-  void readObject(FSTObjectInput in) throws Exception {
-    ObjectId id = (ObjectId) in.readObject(ObjectId.class);
-    this.id = id;
+  static ThreadLocal<Set<ObjectId>> containedObjectIds = ThreadLocal.withInitial(HashSet::new);
+
+  public static List<ObjectId> getAndClearContainedObjectIds() {
```

Comment:
Do we need to call this method after [this line of code](https://github.com/ray-project/ray/blob/745b9d643d8e80100d92421d29db589966d8518d/java/runtime/src/main/java/org/ray/runtime/task/TaskExecutor.java#L98)?